bugbug/bug_features.py

```diff
@@ -209,6 +210,40 @@ def __call__(self, bug, **kwargs):
         return None
 
 
+class delta_nightly_request_merge(single_bug_feature):
+    name = "Time delta between landing of the patch in Nightly and uplift request"
+
+    def __call__(self, bug, **kwargs):
+        for history in bug["history"]:
+            for change in history["changes"]:
+                if not (
+                    change["added"].startswith("approval-mozilla")
+                    and change["added"].endswith("?")
+                ):
+                    continue
+
+                uplift_request_datetime = parser.parse(history["when"])
+
+                landing_comments = Bugzilla.get_landing_comments(
+                    bug["comments"], ["nightly"]
+                )
+
+                # This will help us to find the closest landing before the uplift request
+                landing_time_list = []
+                for landing in landing_comments:
+                    landing_time = parser.parse(landing["comment"]["creation_time"])
+
+                    # Only accept if the uplift is on the future and
+                    # if the landing_time is greater than the calculated now
+                    if uplift_request_datetime >= landing_time:
+                        landing_time_list.append(landing_time)
```

Comment:
It would be more pythonic to use list comprehension here :)

Reply:
Thanks for the advice, it would be same length but a bit more difficult to understand/debug complicated list comprehension; plus, calling parser.parse twice.
here's the code snippet in list comprehension format:


```
landing_time_list = [
                    parser.parse(landing["comment"]["creation_time"]) 
                        for landing in landing_comments 
                        if uplift_request_datetime >= parser.parse(
                            landing["comment"]["creation_time"]
                            ) 
                ]
```
So I would like to keep it as for-loop format.

Reply:
No need to parse the time at this point; you could keep the time in the string format and parse it at the end when you need to calculate the time delta. 
```python
landing_time_list = [
    landing["comment"]["creation_time"]
    for landing in landing_comments
    # Only accept if the uplift is on the future
    if history["when"] >= landing["comment"]["creation_time"]
]
```

In general, the usage of list comprehension is preferred where applicable, but I have no problem either way.